tests/src/Interop/PInvoke/Array/MarshalArrayAsField/LPArrayNative/MarshalArrayByValArrayNative.cpp

```diff
@@ -70,10 +70,64 @@ typedef struct  { BOOL		 arr[ARRAY_SIZE];		}	S_BOOLArray;
 helper function
 ----------------------------------------------------------------------------*/
 
+template<typename T>
+bool IsObjectEquals(T o1, T o2)
+{
+    // T::operator== required.
+    return o1 == o2;
+}
+
+template<>
+bool IsObjectEquals(LPSTR o1, LPSTR o2)
+{
+    size_t cLen1 = strlen(o1);
+    size_t cLen2 = strlen(o2);
+
+    if (cLen1 != cLen2 )
+    {
+        printf("Not equals in %s\n",__FUNCTION__);
+        return false;
+    }
+
+    return strncmp(o1, o2, cLen1) == 0;
+}
+
+template<>
+bool IsObjectEquals(LPCSTR o1, LPCSTR o2)
+{
+    size_t cLen1 = strlen(o1);
+    size_t cLen2 = strlen(o2);
+
+    if (cLen1 != cLen2 )
+    {
+        printf("Not equals in %s\n",__FUNCTION__);
+        return false;
+    }
+
+    return strncmp(o1, o2, cLen1) == 0;
+}
+
+#ifdef _WIN32
+template<>
+bool IsObjectEquals(BSTR o1, BSTR o2)
+{
+    UINT uLen1 = SysStringLen(o1);
+    UINT uLen2 = SysStringLen(o2);
+
+    if (uLen1 != uLen2 )
+    {
+        printf("Not equals in %s\n",__FUNCTION__);
+        return false;
+    }
+
+    return memcmp(o1, o2, uLen1) == 0;
```

Comment:
This is wrong. The `BSTR` is an `OLECHAR *` and `sizeof(OLECHAR) == 2`. The length of the string will be characters so with this logic, you are only comparing the first `uLen1` bytes of the strings. Instead change to `(uLen1 * sizeof(*o1))`.